From 56e476bb962d0e8ade35e0ead69890fa281c7320 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:40 -0400
Subject: [tools] lib/traceevent: Make add_left() return pevent_errno

Message-id: <1408953210-26343-153-git-send-email-jolsa@redhat.com>
Patchwork-id: 88953
O-Subject: [PATCH RHEL7.1 BZ1133083 152/322] tools lib traceevent: Make add_left() return pevent_errno
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit ff533fc058975579dffbb62a731f63911ae714be
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Thu Dec 12 16:36:10 2013 +0900

description
===========
So that it can propagate error properly.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index c08ce59..774c3e4 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -648,7 +648,7 @@ rotate_op_right(struct filter_arg *a, struct filter_arg *b)
  return arg;
 }
 
-static int add_left(struct filter_arg *op, struct filter_arg *arg)
+static enum pevent_errno add_left(struct filter_arg *op, struct filter_arg *arg)
 {
  switch (op->type) {
  case FILTER_ARG_EXP:
@@ -667,11 +667,11 @@ static int add_left(struct filter_arg *op, struct filter_arg *arg)
   /* left arg of compares must be a field */
   if (arg->type != FILTER_ARG_FIELD &&
       arg->type != FILTER_ARG_BOOLEAN)
-   return -1;
+   return PEVENT_ERRNO__INVALID_ARG_TYPE;
   op->num.left = arg;
   break;
  default:
-  return -1;
+  return PEVENT_ERRNO__INVALID_ARG_TYPE;
  }
  return 0;
 }
-- 
1.7.1